Automattic/jetpack
# Dialogue: connect with podcast player via global store

IMPORTANT
Let's hold on until #18349 merge. This branch has been branched off from there.

This PR adds a media player control component and renders it in the block toolbar of the Dialogue block, as long as exists a default media source.

Fixes https://github.com/Automattic/jetpack/issues/18344
Fixes https://github.com/Automattic/jetpack/issues/18286

#### Changes proposed in this Pull Request:

* Dialogue: connect with podcast player via global store

#### Jetpack product discussion


p1610706579156800-slack-

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* In a block editor, adds a `Conversation` block with some `Dialogue` blocks into it.
* Confirm the media control player is disabled.

<img width="400" src="https://user-images.githubusercontent.com/77539/104946369-b1557180-5998-11eb-9b06-d3a78b9d7b73.png" />


* In any place of the editor, adds a podcast player with a valid RSS feed.
* Confirm that the Dialogue block renders the media control player in its toolbar

<img width="509" alt="Screen Shot 2021-01-18 at 11 47 22 AM" src="https://user-images.githubusercontent.com/77539/104929727-04bcc500-5983-11eb-9f1f-3a6f1cb4d9c7.png">

* Confirm you can change playing status by clicking on the play/pause button
* Confirm time is synced between the podcast player and the media control player

* Confirm you play/stop from there, and also the player state is synced with the main player, and vice versa.


#### Proposed changelog entry for your changes:


n/a
